### PR TITLE
Fix calendar issues

### LINKED ISSUE #6562: Calendar view UI issues on small desktop layout
Calendar view issues on small desktop layout

image

 * [x] Use short names for months when there are 2 months, we already use them elsewhere
 * [x] Week number in the column above time. If the time column not displayed (desktop agenda), do not show it
 * [x] Today indicator: try to use darker gray (like icon buttons) and make text inside of it light (see "today" button)
 * [x] Increase minimum size for the nav column so that mini calendar fits into it
 * [x] Mini calendar selected week indicator looks off, margins on the size should not be proportional to the width, they should be fixed, corner radius could be smaller
 * [x] Make sure that "new event" button is aligned with mini calendar, even when nav column has a scrollbar (reserve space for the scrollbar?)

## Original issue

Calendar Weekly view - when the month changes, the week number overrides other information. Language set to Brazilian-Portuguese.
Captura de Tela 2024-02-15 às 16 40 01

## Test notes
- [ ] When there are two months, the short names for months are used.
- [ ] The week number is displayed in the same column as time.
- [ ] The circle highlight for the day is white on dark gray.
- [ ] The mini calendar looks fine at all widths
- [ ] The week highlight in the mini calendar looks good.
- [ ] The "new event" button stays aligned when a scrollbar is present.

### PR DESCRIPTION
**Pull request by paw-hub**
Fixes #6562

### CODE REVIEW

**Comment by charlag** [comment]
1. Header title is not updated anymore
2. Week text does not fit into the column Screenshot from 2024-03-14 11-12-19 we need to figure out some shorter label and/or wrapping

I'm still testing…

File: src/gui/size.ts [hunk 1]
```diff
@@ -62,14 +62,15 @@ export const size = {
...
+	calendar_days_header_height: 25,
 	calendar_hour_width: 80,
 	calendar_hour_width_mobile: 30,
 	calendar_event_margin: 6,
 	calendar_event_margin_mobile: 2,
 	calendar_event_border: 1,
 	calendar_day_event_padding: 2,
 	drawer_menu_width: 44,
-	first_col_min_width: 240,
+	first_col_min_width: 270,
```
Comment (charlag) [inline src/gui/size.ts:73]: this affects all views and probably breaks some breakpoints somewhere (pun not intended). I think I would prefer if we changed it for the calendar view only.

File: src/calendar/view/CalendarView.ts [hunk 2]
```diff
@@ -216,6 +232,7 @@ export class CalendarView extends BaseTopLevelView implements TopLevelView<Calen
 									startOfTheWeek: downcast(locator.logins.getUserController().userSettingsGroupRoot.startOfTheWeek),
 									dragHandlerCallbacks: this.viewModel,
 									isDaySelectorExpanded: this.viewModel.isDaySelectorExpanded(),
+									weekIndicator: this.navConfiguration.week,
```
Comment (charlag) [inline src/calendar/view/CalendarView.ts:235]: I think the only reason to keep weekIndicator in `NavConfiguration` was to share it across the toolbars. We can still pass it in pre-computed but it doesn't have to be in nav config anymore I think

File: src/calendar/view/CalendarView.ts [hunk 3]
```diff
@@ -157,7 +164,7 @@ export class CalendarView extends BaseTopLevelView implements TopLevelView<Calen
 			},
 			ColumnType.Foreground,
 			{
-				minWidth: size.first_col_min_width,
+				minWidth: size.calendar_first_col_min_width,
```
Comment (charlag) [inline src/calendar/view/CalendarView.ts:167]: from changing settings I recall that adjusting one column did require adjusting other columns to keep breakpoints (like `desktop_layout_width` or `two_column_layout_width`) working. Do we also need to do this here or is it fine as-is?

**Review by charlag — CHANGES_REQUESTED** [review]
cool, thanks for addressing the issues!

as discussed, we need to make calendar week fit somehow.

My suggestions:
 * just show the number, maybe inside a similar badge background like we had before
 * otherwise add some short translation like `k.w.` that goes onto its own line

File: src/calendar/view/CalendarView.ts (hunk 3 above)
Comment (rezbyte) [reply src/calendar/view/CalendarView.ts:167]: We have addressed this in our new commit.

**Comment by rezbyte** [comment]
> cool, thanks for addressing the issues!
> 
> as discussed, we need to make calendar week fit somehow.
> 
> My suggestions:
> 
>     * just show the number, maybe inside a similar badge background like we had before
> 
>     * otherwise add some short translation like `k.w.` that goes onto its own line

We took a look at the placement of the week day and it seems fine to us (with @wrdhub). The only case we found were it overflowed was in Italian and Greek but that could be fixed in the translations. It did not look that good with the same background as before so we left it as is. Could you elaborate a bit more on the calendar week fit here perhaps?

**Comment by charlag** [comment]
Screenshot from 2024-03-26 15-20-31

quick test in Russian and the number itself does not fit in there

**Comment by charlag** [comment]
But as you said it can be fixed in a translation

**Comment by rezbyte** [comment]
> Screenshot from 2024-03-26 15-20-31
> 
> quick test in Russian and the number itself does not fit in there

I just tested this and the number is not there to begin with. The Russian translation key is missing the `week` parameter. https://github.com/tutao/tutanota/blob/40610dc1cb6a1a97e24023b6a0d6107425b772b9/src/translations/ru.ts#L1657

Compare this with the same key in English:
https://github.com/tutao/tutanota/blob/40610dc1cb6a1a97e24023b6a0d6107425b772b9/src/translations/en.ts#L1656

**Comment by charlag** [comment]
oops good catch, thank you!

**Review by charlag — APPROVED** [review]

**Comment by charlag** [comment]
Do we need any additional test notes?

**Comment by rezbyte** [comment]
> Do we need any additional test notes?

I can write some.

**Pull request merged**